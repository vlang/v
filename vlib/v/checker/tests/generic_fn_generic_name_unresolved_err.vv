module main

fn func[T](arg string, val T) int {
	return 2
}

fn main() {
	_ := func[T]
	_ := {
		'test': func[T]
	}
	_ := {
		func[T]: 'test'
	}
}
