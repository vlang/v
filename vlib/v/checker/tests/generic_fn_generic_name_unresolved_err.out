vlib/v/checker/tests/generic_fn_generic_name_unresolved_err.vv:8:7: error: a generic fn with generic types, cannot be used outside of another generic fn
    6 | 
    7 | fn main() {
    8 |     _ := func[T]
      |          ~~~~
    9 |     _ := {
   10 |         'test': func[T]
vlib/v/checker/tests/generic_fn_generic_name_unresolved_err.vv:10:11: error: a generic fn with generic types, cannot be used outside of another generic fn
    8 |     _ := func[T]
    9 |     _ := {
   10 |         'test': func[T]
      |                 ~~~~
   11 |     }
   12 |     _ := {
vlib/v/checker/tests/generic_fn_generic_name_unresolved_err.vv:13:3: error: a generic fn with generic types, cannot be used outside of another generic fn
   11 |     }
   12 |     _ := {
   13 |         func[T]: 'test'
      |         ~~~~
   14 |     }
   15 | }
